docs/walkthroughs/secure-ml-pipeline.md

```diff
@@ -0,0 +1,263 @@
+    HTTP/name-0-zkfc.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/name-1-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/name-1-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/name-1-zkfc.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/name-1-zkfc.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/journal-0-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/journal-0-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/journal-1-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/journal-1-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/journal-2-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/journal-2-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/data-0-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/data-0-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/data-1-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/data-1-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs/data-2-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    HTTP/data-2-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL
+    kafka/kafka-0-broker.secure-kafka.autoip.dcos.thisdcos.directory@LOCAL
+    kafka/kafka-1-broker.secure-kafka.autoip.dcos.thisdcos.directory@LOCAL
+    kafka/kafka-2-broker.secure-kafka.autoip.dcos.thisdcos.directory@LOCAL
+    hdfs@LOCAL
+    client@LOCAL
+    ```
+1.  Download the `keytab` file to a local directory.
+1.  Base64 encode the file with `base64 -w 0 {source} > {destination}` (we recommend using `base64` from GNU coreutils, if the version on your machine is different the `-w` flag may need to be replaced with the appropriate flag to disable wrapping.
+1.  Install the DC/OS Enterprise CLI with the command line: `dcos package install --yes dcos-enterprise-cli`   
+1.  Upload the base64 encoded file to the secret store: `dcos security secrets create __dcos_base64__keytab --value-file {destination}`.
+    *   Notice we have uploaded the secret to the root (will be `/__dcos_base64__keytab`)
+1.  You're now ready to install HDFS and Kafka. This can be done using the UI (DC/OS Catalog) or through the CLI as demonstrated here. First make an options.json containing at least (these are the same for both services):
+    ```json
+    {
+      "service": {
+        "kerberos": {
+          "enabled": "true",
+          "kdc": {
+            "hostname":  "<agent_hosting_kdc_or_dns>",
+            "port": "<port_to_communicate_with_kdc>"
+          },
+          "realm": "LOCAL",
+          "keytab_secret": "__dcos_base64__keytab"
+        },
+        "transport_encryption": {
+          "enabled": "true",
+          "allow_plaintext": "false"
+        }
+      }
+    }
+    ```
+    
+## Install Spark and prepare Java Authentication and Authorization Service (JAAS) file for Kafka
+1.  Install Spark including HDFS settings, using the instructions in [installing-secure-spark.md](). Using Kafka from Spark does not require any special installation.
```

Comment:
"Using Kafka from Spark does not require any special installation." 
not sure this is intended here.